### PR TITLE
Design System: Add `Cover` component, `height` to `Box` & text/heading hooks

### PR DESCRIPTION
**Pull request by jxom**
## What changed (plus any additional context for devs)

This PR adds the following features to the design system required for the ENS profiles UI:

- A new `<Cover>` component to handle absolute positioning
- Add `height` prop to `Box`
- Add `useTextStyle` and `useHeadingStyle` hooks

## PoW (screenshots / screen recordings)

nav to https://rds-67cy4a5b3-jxom.vercel.app/

- you can see the `<Cover>` component under the "Cover" menu
- you can see `<Box>` heights under the "Box" menu

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

File: src/design-system/components/Cover/Cover.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Box, BoxProps } from '../Box/Box';
+
+export type CoverProps = {
+  alignHorizontal?: AlignHorizontal;
+  alignVertical?: AlignVertical;
+  children: BoxProps['children'];
+};
+
+/**
+ * @description Renders an absolutely filled container relative to it's parent.
```
Comment (markdalgleish) [inline src/design-system/components/Cover/Cover.tsx:17]:
```suggestion
 * @description Renders an absolutely filled container relative to its parent.
```

Comment (jxom) [reply src/design-system/components/Cover/Cover.tsx:17]: why can't i english

File: src/design-system/components/Box/Box.examples.tsx [hunk 2]
```diff
@@ -129,6 +129,22 @@ export const widths: Example = {
...
+      </Stack>
+    ),
+};
+
+export const heights: Example = {
+  name: 'Heights',
+  Example: () =>
+    source(
+      <Stack space="12px">
+        <Box background="accent" height="30px" padding="24px" />
```
Comment (markdalgleish) [inline src/design-system/components/Box/Box.examples.tsx:142]: These padding props need to be removed, otherwise the Box can't be shorter than 48px.

Comment (jxom) [reply src/design-system/components/Box/Box.examples.tsx:142]: Ah derp. Sorry, that was just copy-pasta.

**Review by markdalgleish — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]
:goberserk:

**Pull request merged**